### PR TITLE
Feature/ml 480 conv layer charge estimation

### PR DESCRIPTION
**Pull request by AlexTeslenko**
Feature/ml 480 conv layer charge estimation

### CODE REVIEW

**Review by vhavrylov-ncube — COMMENTED** [review]
LGTM but let other teammates approve too.

File: libs/ml/src/ops/convolution_1d.cpp [hunk 1]
```diff
@@ -390,6 +390,33 @@ void Convolution1D<TensorType>::ReverseFillOutput(TensorType &gemm_output, Tenso
...
+
+  SizeType input_channels  = this->batch_input_shapes_.front().at(0);
+  SizeType batch_size      = this->batch_input_shapes_.front().at(2);
+  SizeType output_channels = this->batch_input_shapes_.back().at(0);
+  SizeType kernel_height   = this->batch_input_shapes_.back().at(2);
+
+  // output_height=number of stride_size steps over input size
+  SizeType output_height = (this->batch_input_shapes_.front().at(1) -
+                            this->batch_input_shapes_.back().at(2) + this->stride_size_) /
+                           this->stride_size_;
```
Comment (baykaner) [inline libs/ml/src/ops/convolution_1d.cpp]: These lines effectively duplicate a part of the ComputeOutputShape function. You should either abstract these two cases away into a separate private function (e.g. ComputeOutputHeight()), or alternatively you can invoke the ComputeOutputShape function here and use the result.

File: libs/ml/src/ops/convolution_2d.cpp [hunk 2]
```diff
@@ -434,6 +434,39 @@ void Convolution2D<TensorType>::ReverseFillOutput(TensorType &gemm_output, Tenso
...
+
+  // output_height=number of stride_size steps over input height
+  SizeType output_height = (this->batch_input_shapes_.front().at(1) -
+                            this->batch_input_shapes_.back().at(2) + this->stride_size_) /
+                           this->stride_size_;
+
+  // output_width=number of stride_size steps over input width
+  SizeType output_width = (this->batch_input_shapes_.front().at(2) -
+                           this->batch_input_shapes_.back().at(3) + this->stride_size_) /
+                          this->stride_size_;
```
Comment (baykaner) [inline libs/ml/src/ops/convolution_2d.cpp]: same as for conv1d; this code duplication should be fixed

**Review by baykaner — CHANGES_REQUESTED** [review]

**Review by baykaner — DISMISSED** [review]

**Review by vhavrylov-ncube — APPROVED** [review]
LGTM

**Review by baykaner — APPROVED** [review]

**Pull request merged**